File: src/lib/api/shifts.ts
import { invoke } from "@tauri-apps/api/core"

export type ShiftStatus = "OPEN" | "CLOSED"

export type Shift = {
  id: number
  opened_at: string
  closed_at: string | null
  opening_cash: number
  expected_cash: number | null
  closing_cash_counted: number | null
  cash_diff: number | null
  total_sales: number | null
  status: ShiftStatus
  note: string | null
}

export type CloseShiftInput = {
  closing_cash_counted: number
  note: string | null
}

export const getCurrentShift = () => invoke<Shift | null>("get_current_shift")

export const openShift = (openingCash: number, note: string | null) =>
  invoke<Shift>("open_shift", { openingCash, note })

export const closeShift = (shiftId: number, payload: CloseShiftInput) =>
  invoke<Shift>("close_shift", { shiftId, payload })

File: src-tauri/src/domain/shifts.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum ShiftStatus {
    Open,
    Closed,
}

#[derive(Debug, Clone, Serialize)]
pub struct Shift {
    pub id: i64,
    pub opened_at: String,
    pub closed_at: Option<String>,
    pub opening_cash: i64,
    pub expected_cash: Option<i64>,
    pub closing_cash_counted: Option<i64>,
    pub cash_diff: Option<i64>,
    pub total_sales: Option<i64>,
    pub status: ShiftStatus,
    pub note: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CloseShiftInput {
    pub closing_cash_counted: i64,
    pub note: Option<String>,
}
